features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NRF5/source/nRF5xGattServer.cpp

```diff
@@ -451,6 +514,140 @@ void nRF5xGattServer::hwCallback(ble_evt_t *p_ble_evt)
             break;
         }
         case GattServerEvents::GATT_EVENT_WRITE_AUTHORIZATION_REQ: {
+            uint16_t conn_handle = gattsEventP->conn_handle;
+            const ble_gatts_evt_write_t& input_req = gattsEventP->params.authorize_request.request.write;
+            const uint16_t max_size = getBiggestCharacteristicSize();
+
+            // this is a long write request, handle it here.
+            switch (input_req.op) {
+                case BLE_GATTS_OP_PREP_WRITE_REQ: {
+                    // verify that the request is not outside of the possible range
+                    if ((input_req.offset + input_req.len) > max_size) {
+                        sd_ble_gatts_rw_authorize_reply(conn_handle, &write_auth_invalid_offset_reply);
+                        releaseLongWriteRequest(conn_handle);
+                        return;
+                    }
+
+                    // find the write request
+                    long_write_request_t* req = findLongWriteRequest(conn_handle);
+                    if (!req)  {
+                        sd_ble_gatts_rw_authorize_reply(conn_handle, &write_auth_invalid_reply);
+                        releaseLongWriteRequest(conn_handle);
+                        return;
+                    }
+
+                    // initialize the first request by setting the offset
+                    if (req->length == 0) {
+                        req->attr_handle = input_req.handle;
+                        req->offset = input_req.offset;
+                    }
+
+                    // it is disalowed to write backward
+                    if (input_req.offset < req->offset) {
```

Comment:
This condition is checked by the comparison in line 553. No need to check it twice.